paasta_tools/cli/cmds/spark_run.py

```diff
@@ -1003,7 +1003,7 @@ def _auto_add_timeout_for_job(cmd, timeout_job_runtime):
         return cmd
     try:
         timeout_present = re.match(
-            r"^.*timeout[\s]+[\d]*[m|h][\s]+spark-submit .*$", cmd
+            r"^.*timeout[\s]+[\d]+[\.]?[\d]*[m|h][\s]+spark-submit .*$", cmd
```

Comment:
@88manpreet not sure exactly how the timeout value is parsed later on but there are a few edge cases that might cause issues?
`timeout 0.2h` does match
`timeout .2h` does not match
`timeout 2.h` does match
Not sure if any of those will cause issues.

Reply:
Thanks for checking for further cases. 
It should be ok, though.